src/ui/ExportToExcel/ExportToExcel.ts

```diff
@@ -97,21 +105,52 @@ export class ExportToExcel extends Component {
    * Also logs an `exportToExcel` event in the usage analytics.
    */
   public download() {
+    const query = this.buildExcelQuery();
+    this.logger.debug("Performing query following 'Export to Excel' click");
+
+    const endpoint = this.queryController.getEndpoint();
+    this.usageAnalytics.logCustomEvent<IAnalyticsNoMeta>(analyticsActionCauseList.exportToExcel, {}, this.element);
+
+    endpoint.downloadBinary(query).then(content => this.generateExcelFile(content));
+  }
+
+  private buildExcelQuery(): IQuery {
     let query = this.queryController.getLastQuery();
+    query = _.omit(query, ['numberOfResults', 'fieldsToInclude']);
 
-    if (query) {
-      // Remove number of results and fields to include from the last query, because those 2 parameters
-      // should be controlled/modified by the export to excel component.
-      query = _.omit(query, ['numberOfResults', 'fieldsToInclude']);
-      if (this.options.fieldsToInclude) {
-        query.fieldsToInclude = <string[]>this.options.fieldsToInclude;
-      }
-      this.logger.debug("Performing query following 'Export to Excel' click");
-
-      const endpoint = this.queryController.getEndpoint();
-      this.usageAnalytics.logCustomEvent<IAnalyticsNoMeta>(analyticsActionCauseList.exportToExcel, {}, this.element);
-      this._window.location.replace(endpoint.getExportToExcelLink(query, this.options.numberOfResults));
+    if (this.options.fieldsToInclude) {
+      query.fieldsToInclude = <string[]>this.options.fieldsToInclude;
     }
+
+    return {
+      ...query,
+      format: 'xlsx',
+      numberOfResults: this.options.numberOfResults
+    };
+  }
+
+  private generateExcelFile(content: ArrayBuffer) {
+    const blob = new Blob([content], { type: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet' });
+    const url = URL.createObjectURL(blob);
+
+    const a = createAnchor();
+    a.href = url;
+    a.download = this.buildExcelFileName();
+    a.click();
+
+    URL.revokeObjectURL(url);
+  }
+
+  private buildExcelFileName() {
+    const utc = moment().utc();
+    const year = utc.format('YYYY');
+    const month = utc.format('MM');
+    const day = utc.format('DD');
+    const hour = utc.format('HH');
+    const minute = utc.format('mm');
+    const second = utc.format('ss');
+
+    return `query--${year}-${month}-${day}--${hour}-${minute}-${second}.xlsx`;
```

Comment:
The file name is returned by the search-api in the response header.

When we used a GET request opened inside of a window, the browser would automatically check the response header and apply the filename to the downloaded attachement.

However, when using a POST request with the search-ui `performOneCall` and `endpointCaller` functions, these only return the response body. To get the headers, we would need to adjust the code to send this info up the stack.

The other option was to build the name again.

Hmm, writing this out, I think I prefer to send the info up ... 🦆

Reply:
What I understand from the meeting this morning is that jsonP should be no longer supported 🦕 
Maybe it can simplify this bit?

Reply:
Yes, I'd like to do those refactors, but prefer to divide the work across a few PRs. I see:

1. A PR to remove the old code for jsonP and xDomainRequest,
2. A PR to pass the headers up the chain, and removing the above function in favour of the name returned by the search-api.